### DISCUSSION TITLE
[Question][Hw1] Throwing Invalid Argument error for all comparison of boolean operands

### DISCUSSION THREAD

**Discussion by retroinspect** [prosyslab-classroom/cs424-program-reasoning#362] [Q&A]
1. 
According to answer of #26, my submission throws Invalid Argument error for suite:17:cmp_ne_bool and suite:14:cmp_eq_bool, failing some of test cases. 

I thought the interpretation of "integer" referenced in comparison should be applied on ALL comparison operators, include equal and not equal.

Is there any chance for claim? 

2.
Could you provide an example of valid C code to generate such comparison of boolean operands in LLVM given test condition?

This is what I thought:
README limits test cases of all values to be i32 or i1, but as far as I know there is no way to define i1 in C.
So i1 value can be only generated from comparison. 
Therefore there can be no comparison of boolean operands.

**Comment by KihongHeo** [comment]
Does #37 answer your question?
